sources/packages/backend/libs/integrations/src/services/disbursement-schedule/e-cert-generation.service.ts

```diff
@@ -52,585 +21,237 @@ const DISBURSEMENT_SCHEDULES_UPDATE_CHUNK_SIZE = 1000;
 @Injectable()
 export class ECertGenerationService {
   constructor(
-    private readonly studentRestrictionSharedService: StudentRestrictionSharedService,
-    private readonly disbursementOverawardService: DisbursementOverawardService,
-    private readonly systemUsersService: SystemUsersService,
-    private readonly confirmationOfEnrollmentService: ConfirmationOfEnrollmentService,
-    private readonly sfasApplicationService: SFASApplicationService,
-    private readonly disbursementScheduleSharedService: DisbursementScheduleSharedService,
+    @InjectRepository(Application)
+    private readonly applicationRepo: Repository<Application>,
   ) {}
 
   /**
-   * Get the information needed for the e-Cert generation, execute all
-   * Todo: E-cert implementation
-   * calculations needed and return the records ready to generate the e-Cert file.
-   * @param offeringIntensity disbursement offering intensity.
-   * @param entityManager used to execute the commands in the same transaction.
-   * @returns records ready to generate the e-Cert file.
+   * Get eligible disbursements that must have calculations executed prior
+   * to be part of an e-Cert. The data returned will contain the disbursement
+   * entity model to be updated and all supporting information needed to
+   * execute all calculations steps.
+   * @param offeringIntensity offering intensity to retrieve the disbursements.
+   * @returns eligible disbursements to be potentially added to an e-Cert.
    */
-  async prepareDisbursementsForECertGeneration(
+  async getEligibleDisbursements(
```

Comment:
Refactored from `prepareDisbursementsForECertGeneration` removing unnecessary fields that are not needed for the calculations.